doc/source/development/xinference_internals.rst

```diff
@@ -0,0 +1,273 @@
+      xinference-worker = xinference.deploy.cmdline:worker
+
+The command-line ``xinference`` can be refered to code in ``xinference.deploy.cmdline:cli``.
+
+Click
+-----
+We use Click to implement a specific command-line: 
+
+::
+
+  @click.option(
+        "--host",
+        "-H",
+        default=XINFERENCE_DEFAULT_DISTRIBUTED_HOST,
+        type=str,
+        help="Specify the host address for the supervisor.",
+    )
+    @click.option(
+        "--port",
+        "-p",
+        default=XINFERENCE_DEFAULT_ENDPOINT_PORT,
+        type=int,
+        help="Specify the port number for the Xinference web ui and service.",
+    )
+
+For example, the ``xinference-local`` command allows you to define the host address and port.
+
+Actor
+=====
+Xinference is fundamentally based on `Xoscar <https://github.com/xorbitsai/xoscar>`_, our actor framework, 
+which can manage computational resources and Python processes to support scalable and concurrent programming.
+The following is a pseudocode demonstrating how our Worker Actor works, the actual Worker Actor is more complex than this.
+
+::
+
+  import xoscar as xo
+
+  class WorkerActor(xo.Actor):
+    def __init__(self, *args, **kwargs):
+      ... 
+    async def launch_model(self, model_id, n_gpu, ...):  
+      # launch an inference engine, use specific model class to load model checkpoints
+      ...
+    async def list_models(self):  
+      # list models on this actor
+      ...
+    async def terminate_model(self, model_id):  
+      # terminate the model
+      ...
+    async def __post_create__(self):
+      # called after the actor instance is created
+      ...
+    async def __pre_destroy__(self):
+      # called before the actor instance is destroyed
+      ... 
+
+We use the ``WorkerActor`` as an example to illustrate how we build the Xinference. Each actor class
+is a standard Python class that inherits from ``xoscar.Actor``. An instance of this class is a specific actor
+within the actor pool.
+
+- **Define Actor Actions**: Each actor needs to define certain actions or behaviors to accomplish specific tasks.
+  For instance, the model inference ``WorkerActor`` needs to launch the model (``launch_model``), list the models
+  in this actor (``list_models``), terminate a model (``terminate_model``). There are two special methods worth
+  noting. The ``__post_create__`` is invoked before the actor is created, allowing for necessary initializations.
+  The ``__pre_destroy__`` is called after the actor is destroyed, allowing for cleanup or finalization tasks. 
+
+- **Reference Actor and Invoke Methods**: When an actor is created, it yields a reference variable so that other
+  actors can reference it. The actor reference can also be referenced with the address. Suppose the ``WorkerActor``
+  is created and the reference variable is ``worker_ref``,  the ``launch_model`` method of this actor class can
+  be invoked by calling ``worker_ref.launch_model()``.
```

Comment:
Here we can emphasis that even the method is a sync method, when calling it on an actor ref, it will become an async method.